scripts/list-workspace-peer-dependencies.ts

```diff
@@ -0,0 +1,165 @@
+import { getPluginConfiguration } from '@yarnpkg/cli';
+import { Configuration, Project, structUtils } from '@yarnpkg/core';
+import { ppath } from '@yarnpkg/fslib';
+
+/**
+ * Package-level information about a workspace in the monorepo.
+ */
+type WorkspacePackage = {
+  name: string;
+  version: string;
+  isPrivate: boolean;
+  peerDependencies: {
+    name: string;
+    range: string;
+  }[];
+};
+
+/**
+ * A node in the tree that this script builds in order to generate output (or
+ * the whole tree itself).
+ */
+type TreeNode = {
+  name: string;
+  children: TreeNode[];
+};
+
+// Kick off the script.
+main().catch((error) => {
+  console.error(error);
+  process.exitCode = 1;
+});
+
+/**
+ * The entrypoint to this script.
+ *
+ * Uses Yarn to retrieve all of the workspace packages in this monorepo along
+ * with peer dependencies on other workspace packages, then spits out a tree
+ * view.
+ */
+async function main() {
+  const workspacePackages = await getWorkspacePackages();
+  console.log(produceTreeLines(generateTree(workspacePackages)).join('\n'));
+}
+
+/**
+ * Uses Yarn to gather the Yarn workspaces inside of this project (the packages
+ * that are matched by the `workspaces` field inside of `package.json`) and the
+ * peer dependencies on other workspace packages.
+ *
+ * @returns The list of workspaces.
+ */
+async function getWorkspacePackages(): Promise<WorkspacePackage[]> {
+  const cwd = ppath.resolve('..', ppath.cwd());
```

Comment:
This script uses Yarn to build the workspaces list, instead of running `yarn workspaces`. This does something similar to what that command does, and the source is here: https://github.com/yarnpkg/berry/blob/8bfe2d545e986993e4450072bac8b1044e5ebed7/packages/plugin-essentials/sources/commands/workspaces/list.ts#L50